xgboost-predictor/src/main/java/biz/k11i/xgboost/Predictor.java

```diff
@@ -144,6 +144,31 @@ void initObjGbm() {
         return predict(feat, output_margin, 0);
     }
 
+    /**
+     * Generates predictions for given feature vector.
+     *
+     * @param feat          feature vector
+     * @param base_margin   predict with base margin for each prediction
+     * @return prediction values
+     */
+    public float[] predict(FVec feat, float base_margin) {
+        return predict(feat, base_margin, 0);
+    }
+
+    /**
+     * Generates predictions for given feature vector.
+     *
+     * @param feat          feature vector
+     * @param base_margin   predict with base margin for each prediction
+     * @param ntree_limit   limit the number of trees used in prediction
+     * @return prediction values
+     */
+    public float[] predict(FVec feat, float base_margin, int ntree_limit) {
+        float[] preds = predictRaw(feat, base_margin, ntree_limit);
```

Comment:
hmm... do the models that are trained with base_margin necessarily `mparam.base_score == 0`, this is not obvious to me

Reply:
based on the original xgboost code it seems to me that with base_margin used base_score is ignored, will verify with tests

Reply:
ok, thanks @honzasterba